### ISSUE TITLE
Warn or disallow backward incompatible dependency configuration in Gradle metadata

### ISSUE THREAD

**Issue by lacasseio** [gradle/gradle#13611]
### Expected Behavior

Published metadata in future Gradle versions should be consumable by older Gradle versions.

### Current Behavior

Published metadata in Gradle 6.5 to at least 6.2.1 was incompatible with 6.0 because of missing version on for requested capability:

```
* What went wrong:
Could not resolve all artifacts for configuration 'classpath'.
> Could not resolve dev.gradleplugins:gradle-plugin-development:1.0.
  Required by:
      unspecified:unspecified:unspecified > dev.gradleplugins.gradle-plugin-development:dev.gradleplugins.gradle-plugin-development.gradle.plugin:1.0
   > Could not resolve dev.gradleplugins:gradle-plugin-development:1.0.
      > Could not parse module metadata https://plugins.gradle.org/m2/dev/gradleplugins/gradle-plugin-development/1.0/gradle-plugin-development-1.0.module
         > missing 'version' at /variants[3]/dependencies[1]/requestedCapabilities[0]
```

### Context

See https://gradle-community.slack.com/archives/CA7UM03V3/p1593214830327200?thread_ts=1592836025.261000&cid=CA7UM03V3
See https://github.com/gradle-plugins/toolbox/issues/17
See https://github.com/gradle-plugins/toolbox/commit/3ea787f667799cbdf122264ef4b500bbefd9ca39

### Steps to Reproduce 

Try:
```
plugins {
   id 'dev.gradleplugins.java-gradle-plugin' version '1.0'
}
```

### Your Environment

Build scan URL: 
https://gradle.com/s/b3s5i4kdsqele (failing, Gradle 6.0) vs https://scans.gradle.com/s/342hizko34cke (succeed, Gradle 6.2.1)

**Comment by lacasseio** [comment]
Is there any resource that explains breaking change, if any, in the Gradle metadata?

**Comment by ljacomet** [comment]
I believe the issue is #11616 which was fixed in #11691
Both the producer _and_ the parser had a bug. While the fix resolved the forward compatibility - seeing `version: null` is supported, the backward direction was broken. Note that only 6.0 should suffer from that issue.

Maybe we should have increased the Gradle Module Metadata version, so that a better error message is produced.

You would have gotten something like:
`Could not parse module metadata https://plugins.gradle.org/m2/dev/gradleplugins/gradle-plugin-development/1.0/gradle-plugin-development-1.0.module: unsupported format version '1.2' specified in module metadata. This version of Gradle supports format version 1.1.`

And changing the fix today would create its own compatibility issues unfortunately.

What can be done is document this issue and indicate what the fix is: either disable Gradle Module Metadata resolution if possible or upgrade to Gradle 6.1+.

**Comment by lacasseio** [comment]
I see, my point was a warning explaining the backward compatibility issue when version is missing could go a long way. In this specific case, it make the plugin unresolvable because Gradle can't parse the dependencies which then looks bad on my project. It's impossible to test for every single version so having pointers from Gradle as the metadata is generated would be useful to avoid these errors.

As you said, updating the version
